examples/3d/blend_modes.rs

```diff
@@ -0,0 +1,257 @@
+//! This example showcases different blend modes.
+//!
+//! ## Controls
+//!
+//! | Key Binding        | Action                              |
+//! |:-------------------|:------------------------------------|
+//! | `Up` / `Down`      | Increase / Decrease Alpha           |
+//! | `Left` / `Right`   | Rotate Camera                       |
+//! | `Spacebar`         | Toggle Unlit                        |
+//! | `C`                | Randomize Colors                    |
+
+use bevy::prelude::*;
+use rand::random;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(example_control_system)
+        .run();
+}
+
+/// set up a simple 3D scene
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    asset_server: Res<AssetServer>,
+) {
+    let base_color = Color::rgba(0.9, 0.2, 0.3, 1.0);
+    let icosphere_mesh = meshes.add(
+        Mesh::try_from(shape::Icosphere {
+            radius: 0.9,
+            subdivisions: 7,
+        })
+        .unwrap(),
+    );
+
+    // Opaque
+    commands.spawn((
+        PbrBundle {
+            mesh: icosphere_mesh.clone(),
+            material: materials.add(StandardMaterial {
+                base_color,
+                alpha_mode: AlphaMode::Opaque,
+                ..default()
+            }),
+            transform: Transform::from_xyz(-4.0, 0.0, 0.0),
+            ..default()
+        },
+        ExampleControls {
+            unlit: true,
+            color: true,
+        },
+    ));
+
+    // Blend
+    commands.spawn((
+        PbrBundle {
+            mesh: icosphere_mesh.clone(),
+            material: materials.add(StandardMaterial {
+                base_color,
+                alpha_mode: AlphaMode::Blend,
+                ..default()
+            }),
+            transform: Transform::from_xyz(-2.0, 0.0, 0.0),
+            ..default()
+        },
+        ExampleControls {
+            unlit: true,
+            color: true,
+        },
+    ));
+
+    // Premultiplied
+    commands.spawn((
+        PbrBundle {
+            mesh: icosphere_mesh.clone(),
+            material: materials.add(StandardMaterial {
+                base_color,
```

Comment:
I considered doing that originally, but ended up not doing it for two reasons:

- It's useful to understand how premultiplied fades between an add-like effect to a traditional alpha effect as you increase the opacity
- Changing the `base_color` is not enough because of ambient light and the specular highlight, so it still looks kinda weird.

Thinking about it, having three spheres that look exactly the same when the demo starts is _also_ confusing. If I understand it correctly, premultiplied is mostly useful for things like foliage, since you can get alpha without the ugly outline, so maybe a separate demo (comparing it to just blend, or maybe blend and mask) makes more sense?

I can try to come up with something, let me know what y'all think.

Reply:
> Changing the base_color is not enough because of ambient light and the specular highlight, so it still looks kinda weird.

Reading up on the subject I think the alpha needs to be applied to these possibly? As the alpha decreasing means a decrease in light contribution.

I think premultiplied alpha is quite useful for things outside of folliage. In fact nvidia makes mention of better results with premultiplied alpha vs traditional here:
https://developer.nvidia.com/content/alpha-blending-pre-or-not-pre